### PR TITLE
Lesson 33

### PR DESCRIPTION
**Pull request by ArtemNeT13**

### CODE REVIEW

File: src/components/Slider/Slider.js [hunk 1]
```diff
@@ -4,36 +4,30 @@ import BtnSlider from './BtnSlider'
 import data from './dataSlider'
 
 export default function Slider() {
+    const [slideIndex, setSlideIndex] = useState(0)
 
-    const [slideIndex, setSlideIndex] = useState(1)
-
-    // console.log(slideIndex)
+    console.log(slideIndex)
     console.log(setSlideIndex)
```
Comment (romos21) [inline src/components/Slider/Slider.js]: remove logs before pushing

File: src/components/Slider/Slider.js [hunk 2]
```diff
@@ -4,36 +4,30 @@ import BtnSlider from './BtnSlider'
...
-        else if (slideIndex === 1){
-            setSlideIndex(data.length)
-        }
     }
 
     return (
         <div className="container-slider">
-            {data.map((obj, index) => {
+            {
+                data.slice(slideIndex, 4 + slideIndex).map((obj, index) => {
```
Comment (romos21) [inline src/components/Slider/Slider.js]: as I told before, set init sliderCardsLength variable - dont use "4" everywhere

File: src/components/Slider/Slider.js [hunk 3]
```diff
@@ -4,36 +4,30 @@ import BtnSlider from './BtnSlider'
...
                 return (
                     <div className={"slide"}
-                        // key={obj.id}
+                        key={obj.id}
                         // className={slideIndex === index + 1 ? "slide active-anim" : "slide"}
```
Comment (romos21) [inline src/components/Slider/Slider.js]: remove comments before pushing

File: src/components/Slider/Slider.js [hunk 4]
```diff
@@ -4,36 +4,30 @@ import BtnSlider from './BtnSlider'
...
-        }
     }
 
     return (
         <div className="container-slider">
-            {data.map((obj, index) => {
+            {
+                data.slice(slideIndex, 4 + slideIndex).map((obj, index) => {
                 return (
                     <div className={"slide"}
```
Comment (romos21) [inline src/components/Slider/Slider.js]: you can move html you return using map method in separated component calls SliderCard as an example

**Comment by romos21** [comment]
also its better to use svg instead of img (see BtnSlider component)

**Comment by romos21** [comment]
Everything other looks so pretty, GJ!

File: src/components/HomesGuestsLoves/HomesGuestsLoves.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import './HomesGuestsLoves.css';
+import Slider from '../Slider/Slider';
+
+class HomesGuestsLoves extends React.Component {
+    render() {
+        return (
+            <section className={"homes-guests"}>
+                <div className={"section-title"}>
```
Comment (bogomolov-and) [inline src/components/HomesGuestsLoves/HomesGuestsLoves.js]: When you write string to props, you can remove `{}`. But if you have data of another type, `{}` is required
For example:
`<div className="section-title">`

File: src/components/Slider/BtnSlider.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+export default function BtnSlider({ direction, moveSlide }) {
+    return (
+        <button
+            onClick={moveSlide}
+            className={direction === "next" ? "btn-slide next" : "btn-slide prev"}
+        >
+            <img
+                src={direction === "next" ? rightArrow : leftArrow}
+                alt={''}
```
Comment (bogomolov-and) [inline src/components/Slider/BtnSlider.js]: When you write string to props, you can remove `{}`

File: src/components/Slider/BtnSlider.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
```
Comment (bogomolov-and) [inline src/components/Slider/BtnSlider.js]: Use single quotes (`''`) everywhere except props

File: src/components/Slider/Slider.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React, {useState} from 'react';
+import './Slider.css';
+import data from './dataSlider';
+import BtnSlider from './BtnSlider';
+import SliderCard from './SliderCard/SliderCard';
+
+export default function Slider () {
+
```
Comment (bogomolov-and) [inline src/components/Slider/Slider.js]: Remove line break

File: src/components/Slider/Slider.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+                data.slice(slideIndex, picturesOnView + slideIndex).map((obj, index) => {
+                return (
+                    <div className={"slide"} key = {index}>
+                        <SliderCard {...obj}/>
+                    </div>
+                )
+                })
+            }
+            <BtnSlider moveSlide={nextSlide} direction={"next"} />
+            <BtnSlider moveSlide={prevSlide} direction={"prev"}/>
```
Comment (bogomolov-and) [inline src/components/Slider/Slider.js]: When you write string to props, you can remove {}

File: src/components/Slider/Slider.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+            setSlideIndex(slideIndex - 1)
+        }
+    };
+
+    return (
+        <div className="container-slider">
+            {
+                data.slice(slideIndex, picturesOnView + slideIndex).map((obj, index) => {
+                return (
+                    <div className={"slide"} key = {index}>
```
Comment (bogomolov-and) [inline src/components/Slider/Slider.js]: When you write string to props, you can remove `{}` and `key={index}` not `key = {index}`

File: src/components/Slider/Slider.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+import SliderCard from './SliderCard/SliderCard';
+
+export default function Slider () {
+
+    const picturesOnView = 4;
+    const [slideIndex, setSlideIndex] = useState(0)
+    const nextSlide = () => {
+        if (slideIndex < data.length - picturesOnView) {
+            setSlideIndex(slideIndex + 1)
+        };
```
Comment (bogomolov-and) [inline src/components/Slider/Slider.js]: Remove semicolon

**Review by bogomolov-and — APPROVED** [review]

**Review by bogomolov-and — APPROVED** [review]

**Review by bogomolov-and — APPROVED** [review]

**Pull request merged**